EIPS/eip-2539.md

```diff
@@ -0,0 +1,266 @@
+---
+eip: 2539
+title: BLS12-377 curve operations
+author: Alex Vlasov (@shamatar)
+discussions-to: https://ethereum-magicians.org/t/eip-2539-bls12-377-precompile-discussion-thread/4659
+status: Draft
+type: Standards Track
+category: Core
+created: 2020-02-26
+requires: 1109, 2046
+---
+
+## Simple Summary
+This precompile adds operation on BLS12-377 curve (from Zexe paper) as a precompile in a set necessary to *efficiently* perform operations such as BLS signature verification and perform SNARKs verifications. Unique properties of BLS12-377 also later allow to have SNARKs that check BLS12-377 pairing in an efficient way and allow e.g. constant-size BLS signature aggregation.
+
+## Abstract
+
+If `block.number >= X` we introduce *nine* separate precompiles to perform the following operations:
+
+- BLS12_377_G1ADD - to perform point addition on a curve defined over prime field
+- BLS12_377_G1MUL - to perform point multiplication on a curve defined over prime field
+- BLS12_377_G1MULTIEXP - to perform multiexponentiation on a curve defined over prime field
+- BLS12_377_G2ADD - to perform point addition on a curve twist defined over quadratic extension of the base field
+- BLS12_377_G2MUL - to perform point multiplication on a curve twist defined over quadratic extension of the base field
+- BLS12_377_G2MULTIEXP - to perform multiexponentiation on a curve twist defined over quadratic extension of the base field
+- BLS12_377_PAIRING - to perform a pairing operations between a set of *pairs* of (G1, G2) points
+
+Multiexponentiation operation is included to efficiently aggregate public keys or individual signer's signatures during BLS signature verification.
+
+### Proposed addresses table
+
+|Precompile   |Address   |
+|---|---|
+|BLS12_377_G1ADD          | 0x13  |
+|BLS12_377_G1MUL          | 0x14  |
+|BLS12_377_G1MULTIEXP     | 0x15  |
+|BLS12_377_G2ADD          | 0x16  |
+|BLS12_377_G2MUL          | 0x17  |
+|BLS12_377_G2MULTIEXP     | 0x18  |
+|BLS12_377_PAIRING        | 0x19  |
+
+## Motivation
+Motivation of this precompile is to add a cryptographic primitive that allows to get 120+ bits of security for operations over pairing friendly curve compared to the existing BN254 precompile that only provides 80 bits of security. In addition it allows efficient one-time recursive proof aggregations, e.g. proofs about existence of BLS12-377 based signature.
+
+## Specification
+
+Curve parameters:
+
+BLS12-377 curve is fully defined by the following set of parameters (coefficient `A=0` for all BLS12 curves):
```

Comment:
```suggestion
[BLS12-377](../assets/eip-2539/bls12-377.pdf) curve is fully defined by the following set of parameters (coefficient `A=0` for all BLS12 curves):
```
This would be a good spot to have a relative link out to an attachment of the BLS12-377 paper. (note: if you accept this comment, make sure to actually add the paper as an asset!)